.github/workflows/kerneltest.yml

```diff
@@ -0,0 +1,237 @@
+        with:
+          go-version-file: .go-version
+
+      - name: Initialize and update libbpf submodule
+        run: git submodule init && git submodule update
+
+      - name: Install libbpf dependencies
+        run: |
+          apt-get update -y
+          apt-get install -yq lld libelf-dev zlib1g-dev libelf-dev:arm64 zlib1g-dev:arm64 pkg-config
+
+      - name: Get branch name
+        shell: bash
+        run: echo "GITHUB_BRANCH_NAME=${GITHUB_REF#refs/*/}" >> $GITHUB_ENV
+
+      - uses: actions/download-artifact@9bc31d5ccc31df68ecc42ccf4149144866c47d8a # v3.0.2
+        with:
+          name: ebpf-object-file-container
+          path: bpf/out
+
+      - name: List downloaded files
+        shell: bash
+        run: |
+          ls -lR bpf
+          mkdir -p pkg/profiler/cpu/bpf/programs/objects
+          cp -r bpf/out/* pkg/profiler/cpu/bpf/programs/objects
+          rm pkg/profiler/cpu/bpf/programs/objects/*/pid_namespace.bpf.o
+          mkdir -p pkg/contained/bpf
+          cp -r bpf/out/* pkg/contained/bpf
+          find pkg/contained/bpf/*/ -type f | grep -v "pid_namespace.bpf.o" | xargs -I{} bash -c "rm {}"
+
+      - name: Run Goreleaser
+        run: |
+          goreleaser release --clean --skip-validate --skip-publish --snapshot --debug -f .goreleaser-test-binaries.yml
+        env:
+          GORELEASER_CURRENT_TAG: "${{ env.goreleaser_current_tag }}"
+
+      - name: Archive generated artifacts
+        uses: actions/upload-artifact@a8a3f3ad30e3422c9c7b888a15615d19a852ae32 # v3.1.3
+        with:
+          name: parca-agent-dist-container
+          if-no-files-found: error
+          path: |
+            goreleaser/dist
+            !goreleaser/dist/*.txt
+
+  run-kerneltest:
+    name: Run kerneltest
+    runs-on: ubuntu-latest
+    needs: build-binaries
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
+        with:
+          fetch-depth: 0
+
+      # The "dist" path intended here is parca-agent/dist, not goreleaser/dist
+      - uses: actions/download-artifact@9bc31d5ccc31df68ecc42ccf4149144866c47d8a # v3.0.2
+        with:
+          name: parca-agent-dist-container
+          path: dist
+
+      - name: Set up Go
+        uses: actions/setup-go@93397bea11091df50f3d7e59dc26a7711a8bcfbe # v4.1.0
+        with:
+          go-version-file: .go-version
+
+      - name: Install dependencies
+        run: |
+          sudo apt-get update -y
+          sudo apt -y install curl qemu-system-x86 qemu-system-aarch64
+
+      - name: List downloaded files
+        shell: bash
+        run: |
+          ls -lR dist
+
+      # TODO(sylfrena): add this step as a target in Makefile for local dev
+      - name: Build initramfs
+        run: |
+          # Copy amd64 test
+          cp ./dist/parca-agent-test-amd64_linux_amd64_v1/kerneltest/cpu.test ./kerneltest/amd64/cpu.test
+
```

Comment:
Instructions here ideally should be part of the `make initramfs` target in out Makefile but since we might be switching to Magefiles or another build system as part of https://github.com/parca-dev/parca-agent/pull/2368#issuecomment-1862823498 we can do this as follow up if we need to.